### DISCUSSION TITLE
4번째 미팅 때 논의하고 싶은 내용

### DISCUSSION THREAD

**Discussion by Joseph-Cha** [Joseph-Cha/JustFight_Client#11] [Q&A] [answered]
1. 요청하신 작은 단위의 분석 및 설계에 대한 구체적인 내용
   - 작은 단위의 분석 및 설계를 하기 전에 먼저 큰 단위(클래스 구조)에 대한 설계를 작성 후 행위 구조에서 작은 단위에 대한 설계를 하는 방향으로 생각하고 있었는데 이렇게 가도 괜찮을지?
   - 아직 좀 막연한 느낌이 들어서 구체적으로 한가지 사례가 있으면 좋을 것 같습니다\
   ex. 캐릭터 이동과 관련된 작은 단위 분석을 어떻게 하면 좋을지??

**Comment by jongfeel** [comment, marked as answer]
### 큰 단위가 클래스 구조를 뜻하지는 않음

여기서 얘기하는 단위는 어떤 한 줄의 요구사항일 수 있고
- 캐릭터는 무기를 장착할 수 있다
UI(화면) 단위: 한 화면을 표현하는 UI 적인 요소일 수도 있고
기능 단위: system call, library 사용, network를 통한 API 호출일 수 있음
객체 단위: 한 캐릭터, 한 맵, 한 몬스터, 한 무기에 대한 속성 그리고 행위에 대한 것, 객체들 끼리의 속성 요청 방식 (이벤트, 메소드)

### 캐릭터의 이동이라면?

터치 인터페이스와 함께 이동과 관련된 라이브러리를 찾아봐도 되고
방향에 대한 값을 찾아내서 -1.0 ~ 1.0 사이의 값으로 변환

여기서 요구사항이 뽑히면 됨
- 캐릭터는 화면의 방향 조이스틱 UI로 캐릭터 이동을 상하좌우로 할 수 있다.
